neurom/morphmath.py

```diff
@@ -483,14 +481,8 @@ def principal_direction_extent(points):
     # principal components
     _, eigv = pca(points)
 
-    extent = np.zeros(3)
-
-    for i in range(eigv.shape[1]):
-        # orthogonal projection onto the direction of the v component
-        scalar_projs = np.sort(np.array([np.dot(p, eigv[:, i]) for p in points]))
-        extent[i] = scalar_projs[-1]
-
-        if scalar_projs[0] < 0.:
-            extent -= scalar_projs[0]
-
-    return extent
+    # for each eigenvector calculate the range of the scalar projections of all points on it
+    return np.fromiter(
+        (np.ptp(np.inner(points, eigv[:, i])) for i in range(eigv.shape[1])),
```

Comment:
would this work?
`return np.ptp(point.dot(eigv.T))`

Reply:
yeah, but it would be unreadable :)

Reply:
However, it's a good point that `points.dot` performs matrix mult correctly wrt eigv.